internal/beater/otlp/stacktrace_processor.go

```diff
@@ -0,0 +1,64 @@
+package otlp
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/elastic/apm-data/model/modelpb"
+	"github.com/elastic/apm-server/internal/r8"
+)
+
+type MapFetcher interface {
+	Fetch(ctx context.Context, name, version string) ([]byte, error)
+}
+
+type CrashDeobfuscator interface {
+	Deobfuscate(stacktrace *[]*modelpb.StacktraceFrame, mapFile io.Reader) error
+}
+
+type StacktraceProcessor struct {
+	fetcher      MapFetcher
+	deobfuscator CrashDeobfuscator
+}
+
+// NewStacktraceProcessor returns a StacktraceProcessor.
+func NewStacktraceProcessor(fetcher MapFetcher) *StacktraceProcessor {
+	return &StacktraceProcessor{fetcher, crashDeobfuscatorFunc(r8.Deobfuscate)}
```

Comment:
May be good to not bake in r8 here and have the caller compose it instead. Removes package dependency to r8, at least in this file.
```suggestion
func NewStacktraceProcessor(fetcher MapFetcher, deobfuscator CrashDeobfuscator) *StacktraceProcessor {
	return &StacktraceProcessor{fetcher, deobfuscator}
```